### PR TITLE
 create open intern position

### PR DESCRIPTION
**Pull request by BrustureanElena**

### CODE REVIEW

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 1]
```diff
@@ -83,4 +85,11 @@ public OpenInternPositionController(OpenInternPositionService openInternPosition
         return openInternPositionService.findOpenInternPositionDTOByCompany(company);
 
     }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.OK)
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: Please use `HttpStatus.CREATED` instead.

File: src/main/java/com/sintern/service/OpenInternPositionServiceImpl.java [hunk 2]
```diff
@@ -24,6 +24,17 @@ public OpenInternPositionServiceImpl(OpenInternPositionRepository openInternPosi
         this.openInternPositionRepository = openInternPositionRepository;
     }
 
+    @Override
+    public void addOpenInternPosition(int availablePositions, String department, String description, String name, Company company) {
+        OpenInternPosition openInternPosition = new OpenInternPosition();
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/OpenInternPositionServiceImpl.java]: Please delegate this to a `Transformer` (Request -> Model Object, i.e. `OpenInternPosition`) and pass only that object to this method.

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 3]
```diff
@@ -83,4 +85,11 @@ public OpenInternPositionController(OpenInternPositionService openInternPosition
...
+    public void createOpenInternPosition(@RequestBody CreateOpenInternPositionRequest createOpenInternPositionRequest) {
+        openInternPositionService.addOpenInternPosition(createOpenInternPositionRequest.getAvailablePositions(),
+                createOpenInternPositionRequest.getDepartment(), createOpenInternPositionRequest.getDescription(), createOpenInternPositionRequest.getName(), companyService.findByID(createOpenInternPositionRequest.getCompanyID()));
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: `companyService.findByID(createOpenInternPositionRequest.getCompanyID())` this should be used in the `OpenInternPositionServiceImpl`. Thus, pass as arguments only the **transformed object** and the `CompanyId`.

File: src/main/java/com/sintern/api/OpenInternPositionController.java (hunk 1 above)
Comment (BrustureanElena) [reply src/main/java/com/sintern/api/OpenInternPositionController.java]: Done

File: src/main/java/com/sintern/service/OpenInternPositionServiceImpl.java (hunk 2 above)
Comment (BrustureanElena) [reply src/main/java/com/sintern/service/OpenInternPositionServiceImpl.java]: Done

File: src/main/java/com/sintern/api/OpenInternPositionController.java (hunk 3 above)
Comment (BrustureanElena) [reply src/main/java/com/sintern/api/OpenInternPositionController.java]: Done

**Pull request merged**